api/routes/application.py

```diff
@@ -77,37 +85,94 @@ def create_application(
     return application
 
 
-@application_router.put("/{application_id}", status_code=status.HTTP_200_OK, response_model=s.ApplicationOut)
+@application_router.put(
+    "/{application_id}",
+    status_code=status.HTTP_200_OK,
+    response_model=s.ApplicationPutOut,
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Application not found"},
+    },
+)
 def update_application(
     application_id: int,
-    data: s.ApplicationPut,
+    data: s.ApplicationPutIn,
     db: Session = Depends(get_db),
     current_user: m.User = Depends(get_current_user),
 ):
-    application: m.Application | None = db.scalar(sa.select(m.Application).where(m.Application.id == application_id))
+    if not data.status:
```

Comment:
This check is redundant. Just make sure that status is not optional field in `ApplicationPutIn` Pydantic will do the validation for you